File: core/infrastructure/src/main/java/com/gonezo/multiplatform/audioextraction/infrastructure/llm/InMemoryExtractionTelemetry.java
package com.gonezo.multiplatform.audioextraction.infrastructure.llm;

import java.util.ArrayList;
import java.util.List;

public final class InMemoryExtractionTelemetry implements ExtractionTelemetry {
  private final List<String> entries = new ArrayList<>();

  @Override
  public void llmCall(String requestId, long durationMs, int inputSize, boolean success) {
    entries.add(requestId + ":" + durationMs + ":" + inputSize + ":" + success);
  }

  public List<String> entries() {
    return List.copyOf(entries);
  }
}

File: core/src/main/kotlin/com/gonezo/audioextraction/infrastructure/llm/InMemoryExtractionTelemetry.kt
package com.gonezo.audioextraction.infrastructure.llm

class InMemoryExtractionTelemetry : ExtractionTelemetry {
  private val entries = mutableListOf<String>()

  override fun llmCall(requestId: String, durationMs: Long, inputSize: Int, success: Boolean) {
    entries.add("$requestId:$durationMs:$inputSize:$success")
  }

  fun entries(): List<String> = entries.toList()
}
